robotmpcs/models/mpcModel.py

```diff
@@ -1,87 +1,54 @@
+import os
 import casadi as ca
 from dataclasses import dataclass
 from forwardkinematics.fksCommon.fk import ForwardKinematics
 from forwardkinematics.urdfFks.generic_urdf_fk import GenericURDFFk
 import numpy as np
+import sys
+sys.path.append("../")
+sys.path.append("")
+from examples.helpers import load_forces_path
+from forwardkinematics.urdfFks.generic_urdf_fk import GenericURDFFk
+load_forces_path()
 import forcespro
 import yaml
-from shutil import move
+from shutil import move, rmtree
 from glob import glob
+from robotmpcs.models.mpcBase import MpcBase
+from robotmpcs.models.objectives.goal_mpc_objective import GoalMpcObjective
+from robotmpcs.models.inequalities.InequalityManager import InequalityManager
 
 
-def diagSX(val, size):
-    a = ca.SX(size, size)
-    for i in range(size):
-        a[i, i] = val[i]
-    return a
-
 
-@dataclass
-class MpcConfiguration:
-    time_horizon: int
-    time_step: float
-    weights: dict
-    slack: bool
-    interval: int
-    number_obstacles: int
-    model_name: str
-    n: int
-    name: str = 'mpc'
-    debug: bool = False
 
 
-@dataclass
-class RobotConfiguration:
-    collision_links: list
-    selfCollision: dict
-    urdf_file: str
-    root_link: str
-    end_link: str
-    base_type: str
 
 
 
-class MpcModel(object):
+class MpcModel(MpcBase):
     def __init__(self, initParamMap=True, **kwargs):
-        self._config = MpcConfiguration(**kwargs['mpc'])
-        self._robot_config = RobotConfiguration(**kwargs['robot'])
-        with open(self._robot_config.urdf_file, 'r') as f:
-            urdf = f.read()
-        self._modelName = self._config.model_name
-        self._fk = GenericURDFFk(
-            urdf,
-            self._robot_config.root_link,
-            self._robot_config.end_link,
-            base_type=self._robot_config.base_type,
-        )
-        self._m = 3
-        self._dt = self._config.time_step
-        if self._robot_config.base_type == 'holonomic':
-            self._n = self._fk.n() 
-            self._nx = 2 * self._n
-            self._nu = self._n
-        elif self._robot_config.base_type == 'diffdrive':
-            self._n = self._fk.n() + 3
-            self._nx = 2 * self._n + 2
-            self._nu = 2 + self._fk.n()
-        self._ns = 0
-        self._n_obst = 0
-        self._m_obst = 3
-        self._pairs = []
-        self._N = self._config.time_horizon
+
+        super().__init__(**kwargs)
+        self._kwargs = kwargs
+
         if initParamMap:
             self._limits = {
                 "x": {"low": np.ones(self._nx) * -100, "high": np.ones(self._nx) * 100},
                 "u": {"low": np.ones(self._nu) * -100, "high": np.ones(self._nu) * 100},
                 "s": {"low": np.zeros(1), "high": np.ones(1) * np.inf},
             }
             self.initParamMap()
+        inequality_list = ['obstacle_avoidance', 'speed_limits', 'joint_limits', 'input_limits', 'self_collision']
```

Comment:
`speed` and `vel` are now used interchangable. We should be consistent on that.

Reply:
This line is not required anymore but I changed `speed` to `vel` to be consistent.